### DISCUSSION TITLE
Merlin Commander Ultimate - Firmware version to use

### DISCUSSION THREAD

**Discussion by firefly2k3** [Kaldek/rat-ratgdo#112] [General]
Hi, I have a Merlin Commander Ultimate wall mounted  jackshaft opener (MJ3800MYQ). I finished the build and flashed with the 2.5i ESPHOME firmware. It shows up in HA but after wiring it up to the GDO, it cannot control anything and doesn't register any state changes except for the OBS sensor. If I obstruct the IR beams, the state changes to "Problem" and clears to "OK" when obstruction is removed.

Here's the completed board:
![20241016_122558](https://github.com/user-attachments/assets/9e07e32a-2067-461b-b8fa-8e23cb14ca1f)
![20241016_132623](https://github.com/user-attachments/assets/5c22136c-2c98-4bb7-b16c-ce0fd32a769f)

I have battery backup on the GDO so used a step down to provide a 3.3v power source to the rat-ratgdo.
I've connected OBS to 5, CTRL to 1 and GND to 3.
![image](https://github.com/user-attachments/assets/064fadb2-b051-4eff-9d14-eee4908be366)

Anyone has the same model GDO and had success?

**Comment by firefly2k3** [comment]
Anyone?

**Reply by foreverimagining** [reply]
Quick question: which protocol did you use the firmware of?

Jackshaft openers often have yellow learn buttons that make you think it is Security+2.0, but they actually run Security+1.0, which has always been a rather trippy contradiction. The firmware you flash needs to match the correct protocol.

**Reply by firefly2k3** [reply]
I've flashed 2.5i Security+ 2.0 firmware.
Since it was not working and I read that some jackshaft openers use Security+ 1.0, I tried that too. But that gives me continuous error log "Security+ 1.0 controller not found". So I flashed back 2.0. 

I gave up and left it connected for a few days. Now I cannot get to the ratgdo web console at all. Physically inspecting the board shows that blue light keeps flashing every few seconds indicating it's in a boot loop of some sorts. Reset button brings it back momentarily but goes into loop again. 

I'm thinking faulty D1 Mini... Can't think of anything else as I've tried everything including powering from a usb-c adpater instead of the buck converter. Have ordered a couple more D1 minis. Will see if that solves it once it arrives.
